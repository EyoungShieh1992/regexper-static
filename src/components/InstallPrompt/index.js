import React, { useState, useEffect, useCallback } from 'react';
import { Trans } from 'react-i18next';

const InstallPrompt = () => {
  const [ installPrompt, updateInstallPrompt ] = useState(null);

  const handleInstall = useCallback(async event => {
    event.preventDefault();

    try {
      installPrompt.prompt();
      await installPrompt.userChoice;
    }
    catch {
      // User cancelled install
    }

    updateInstallPrompt(null);
  }, [installPrompt, updateInstallPrompt]);

  useEffect(() => {
    window.addEventListener('beforeinstallprompt', updateInstallPrompt);

    return () => {
      window.removeEventListener('beforeinstallprompt', updateInstallPrompt);
    };
  });

  if (!installPrompt) {
    return null;
  }

  return <a href="#install"
    data-testid="install"
    onClick={ handleInstall }
  >
    <Trans>Add to Home Screen</Trans>
  </a>;
};

export default InstallPrompt;
