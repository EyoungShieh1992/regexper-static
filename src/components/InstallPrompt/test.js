import React from 'react';
import { render, fireEvent } from 'react-testing-library';

import InstallPrompt from 'components/InstallPrompt';

describe('InstallPrompt', () => {
  test('rendering', () => {
    const { asFragment } = render(
      <InstallPrompt />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('rendering after an install prompt has been requested', () => {
    const { asFragment } = render(
      <InstallPrompt />
    );
    expect(asFragment()).toMatchSnapshot();

    const event = new Event('beforeinstallprompt', {
      prompt: jest.fn()
    });
    fireEvent(window, event);

    expect(asFragment()).toMatchSnapshot();
  });

  test('removing event listener', () => {
    jest.spyOn(window, 'addEventListener');
    jest.spyOn(window, 'removeEventListener');

    const { unmount } = render(
      <InstallPrompt />
    );

    unmount();

    expect(window.removeEventListener).toHaveBeenCalledWith(
      'beforeinstallprompt',
      expect.any(Function));
  });
});
