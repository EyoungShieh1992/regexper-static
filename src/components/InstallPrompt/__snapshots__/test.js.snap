// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstallPrompt rendering 1`] = `<DocumentFragment />`;

exports[`InstallPrompt rendering after an install prompt has been requested 1`] = `<DocumentFragment />`;

exports[`InstallPrompt rendering after an install prompt has been requested 2`] = `
<DocumentFragment>
  <a
    data-testid="install"
    href="#install"
  >
    <span
      data-component="Trans"
      data-props="{}"
    >
      Add to Home Screen
    </span>
  </a>
</DocumentFragment>
`;
